File: src/main/java/com/thebluehats/server/game/managers/enchants/processedevents/CastedEntityDamageByEntityEvent.java
package com.thebluehats.server.game.managers.enchants.processedevents;

import org.bukkit.entity.Player;
import org.bukkit.event.entity.EntityDamageByEntityEvent;

public class CastedEntityDamageByEntityEvent extends CastedEvent<EntityDamageByEntityEvent> {
    private final Player damager;
    private final Player damagee;

    public CastedEntityDamageByEntityEvent(EntityDamageByEntityEvent event, Player damager, Player damagee) {
        super(event);

        this.damager = damager;
        this.damagee = damagee;
    }

    public Player getDamager() {
        return damager;
    }

    public Player getDamagee() {
        return damagee;
    }
}

File: src/main/java/com/thebluehats/server/game/managers/enchants/processedevents/CastedEntityDamageByEntityEvent.kt
package com.thebluehats.server.game.managers.enchants.processedevents

import org.bukkit.entity.Player
import org.bukkit.event.entity.EntityDamageByEntityEvent

open class CastedEntityDamageByEntityEvent(
    event: EntityDamageByEntityEvent,
    val damager: Player,
    val damagee: Player
) : CastedEvent<EntityDamageByEntityEvent>(event)